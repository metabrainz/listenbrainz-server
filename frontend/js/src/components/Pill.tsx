import * as React from "react";

type PillProps = {
  active?: boolean;
  type?: "primary" | "secondary";
  style?: React.CSSProperties;
  className?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  [key: string]: any;
};

export default function Pill(props: React.PropsWithChildren<PillProps>) {
  const {
    active,
    children,
    type,
    style: propStyle,
    className: propClassName = "",
    ...buttonProps
  } = props;

  return (
    <button
      type="button"
      style={propStyle}
      {...buttonProps}
      className={`pill ${type === "secondary" ? "secondary" : ""} ${
        active ? "active" : ""
      } ${propClassName}`}
    >
      {children}
    </button>
  );
}
