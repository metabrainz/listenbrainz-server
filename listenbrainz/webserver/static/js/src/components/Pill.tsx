import * as React from "react";

type PillProps = {
  active?: boolean;
  type?: "primary" | "secondary";
  style?: React.CSSProperties;
  className?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  [key: string]: any;
};

export default function Pill(props: React.PropsWithChildren<PillProps>) {
  const { active, children, type, style: propStyle, ...buttonProps } = props;

  let background = "#EB743B";
  if (type === "secondary") {
    background = "#353070";
  }

  let style: React.CSSProperties = {
    borderRadius: "24px",
    outline: "none",
    padding: "3px 12px",
    margin: "2px 6px 12px 6px",
    boxSizing: "border-box",
  };

  if (active) {
    style = {
      background,
      border: `2px solid ${background}`,
      fontWeight: 700,
      color: "white",
      ...style,
      ...propStyle,
    };
  } else {
    style = {
      background: "transparent",
      border: "2px solid #BBBBBB",
      color: "#BBBBBB",
      ...style,
      ...propStyle,
    };
  }
  return (
    <button type="button" style={style} {...buttonProps}>
      {children}
    </button>
  );
}
