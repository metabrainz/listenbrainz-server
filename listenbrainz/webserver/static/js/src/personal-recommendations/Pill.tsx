import * as React from "react";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
import ListenControl from "../listens/ListenControl";

export type PillProps = {
  title: string;
  closeAction?: (event: React.MouseEvent) => void;
  closeButton?: Boolean;
};

const Pill = (props: PillProps) => {
  const { title, closeAction, closeButton } = props;
  return (
    <div className="pill">
      <div>
        <span>{title}</span>
        {closeButton && (
          <ListenControl text="" icon={faTimesCircle} action={closeAction} />
        )}
      </div>
    </div>
  );
};

export default Pill;
