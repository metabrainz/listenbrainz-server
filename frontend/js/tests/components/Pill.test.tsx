import * as React from "react";
import { mount } from "enzyme";

import Pill from "../../src/components/Pill";

describe("Pill", () => {
  it("renders correctly for primary active", () => {
    const wrapper = mount(<Pill active type="primary" />);
    expect(wrapper.getDOMNode()).toContainHTML(
      '<button type="button" class="pill  active "></button>'
    );
  });

  it("renders correctly for secondary active", () => {
    const wrapper = mount(<Pill active type="secondary" />);
    expect(wrapper.getDOMNode()).toContainHTML(
      '<button type="button" class="pill secondary active "></button>'
    );
  });

  it("renders correctly for inactive", () => {
    const wrapper = mount(<Pill />);
    expect(wrapper.getDOMNode()).toContainHTML(
      '<button type="button" class="pill   "></button>'
    );
  });
});
