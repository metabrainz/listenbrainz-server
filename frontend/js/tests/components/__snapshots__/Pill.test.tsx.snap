// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill renders correctly for inactive 1`] = `
<Pill>
  <button
    style={
      Object {
        "background": "transparent",
        "border": "2px solid #8d8d8d",
        "borderRadius": "24px",
        "boxSizing": "border-box",
        "color": "#8d8d8d",
        "margin": "2px 6px 12px 6px",
        "outline": "none",
        "padding": "3px 12px",
      }
    }
    type="button"
  />
</Pill>
`;

exports[`Pill renders correctly for primary active 1`] = `
<Pill
  active={true}
  type="primary"
>
  <button
    style={
      Object {
        "background": "#eb743b",
        "border": "2px solid #eb743b",
        "borderRadius": "24px",
        "boxSizing": "border-box",
        "color": "#fff",
        "fontWeight": 700,
        "margin": "2px 6px 12px 6px",
        "outline": "none",
        "padding": "3px 12px",
      }
    }
    type="button"
  />
</Pill>
`;

exports[`Pill renders correctly for secondary active 1`] = `
<Pill
  active={true}
  type="secondary"
>
  <button
    style={
      Object {
        "background": "#353070",
        "border": "2px solid #353070",
        "borderRadius": "24px",
        "boxSizing": "border-box",
        "color": "#fff",
        "fontWeight": 700,
        "margin": "2px 6px 12px 6px",
        "outline": "none",
        "padding": "3px 12px",
      }
    }
    type="button"
  />
</Pill>
`;
